### PR TITLE
Project Report GSoC 2016

### PR DESCRIPTION
**Pull request by pulkit4tech**
Project Report which is to be submitted for GSoC 2016 `Final Evaluation`

### CODE REVIEW

**Comment by kenichi** [comment]
i don't foresee us adding a "yaml" backend for session support, but maybe a "memcached" one. :)

File: reports/2016-New_Reel_Features_Project_Report.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+```
+if multipart type
+      return Hash :
+      {
+       key1 => {
+         :data => Tempfile object (contains parsed data),
+         :complete => true/false,
+         :part => part object (header info etc)
+       },
+       key2 => {...}
```
Comment (kenichi) [inline reports/2016-New_Reel_Features_Project_Report.md]: can you add some description about the keys?

File: reports/2016-New_Reel_Features_Project_Report.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+**Abstract**
+
+Session Handler can be turned `on` into Reel by requiring `reel/session` module in application.Custom Session Configuration can be passed during server initialization as shown in this [example](https://github.com/pulkit4tech/reel/blob/final_sessions/examples/custom_session.rb) or else `DEFAULT` Session configuration will be used like in this [example](https://github.com/pulkit4tech/reel/blob/final_sessions/examples/hello_world_session.rb). After Header from request have been parsed, `Session Hash Value` is exposed depending on uuid recieved from request which can be accessed using `request.session`. Session value is added/updated into `Store` during `response` and corresponding uuid is set into `Cookie`. `Celluloid::Timers` are used for cleaning Session Values from Store after expiry and are `reset` if Session Value is updated, keeping Store clean and consistent. 
+
+**Extension**
+
+Currently we are using `Celluloid Concurrent Hash` for `Store`, but plan is to extend it by implementing multiple stores/supporting drivers like `Hash with mutex`, `Redis`, `Yaml`, `Mongodb` etc which can be used based on Application context.
```
Comment (kenichi) [inline reports/2016-New_Reel_Features_Project_Report.md]: i don't foresee us adding a "yaml" backend for session support, but maybe a "memcached" one. 😄

Comment (pulkit4tech) [inline reports/2016-New_Reel_Features_Project_Report.md]: Yeah @kenichi , updating :smile:

**Comment by pulkit4tech** [comment]
@kenichi : is it good now or should I add description for `key` like `Here keys are part.name(which is key in`key=value`for POST request) for different file upload`? :smiley:

**Pull request merged**